source/frontend/src/components/common/mapFSM/useLoactionFeatureLoader.tsx

```diff
@@ -0,0 +1,101 @@
+import { Feature, FeatureCollection, Geometry } from 'geojson';
+import { LatLngLiteral } from 'leaflet';
+import { useCallback } from 'react';
+
+import { useAdminBoundaryMapLayer } from '@/hooks/repositories/mapLayer/useAdminBoundaryMapLayer';
+import { useFullyAttributedParcelMapLayer } from '@/hooks/repositories/mapLayer/useFullyAttributedParcelMapLayer';
+import { useLegalAdminBoundariesMapLayer } from '@/hooks/repositories/mapLayer/useLegalAdminBoundariesMapLayer';
+import { useMapProperties } from '@/hooks/repositories/useMapProperties';
+import { MOT_DistrictBoundary_Feature_Properties } from '@/models/layers/motDistrictBoundary';
+import { MOT_RegionalBoundary_Feature_Properties } from '@/models/layers/motRegionalBoundary';
+import { WHSE_Municipalities_Feature_Properties } from '@/models/layers/municipalities';
+import { PMBC_FullyAttributed_Feature_Properties } from '@/models/layers/parcelMapBC';
+import { PIMS_Property_Location_View } from '@/models/layers/pimsPropertyLocationView';
+
+export interface LocationFeatureDataset {
+  location: LatLngLiteral;
+  pimsFeature: Feature<Geometry, PIMS_Property_Location_View> | null;
+  parcelFeature: Feature<Geometry, PMBC_FullyAttributed_Feature_Properties> | null;
+  regionFeature: Feature<Geometry, MOT_RegionalBoundary_Feature_Properties> | null;
+  districtFeature: Feature<Geometry, MOT_DistrictBoundary_Feature_Properties> | null;
+  municipalityFeature: Feature<Geometry, WHSE_Municipalities_Feature_Properties> | null;
+}
+
+const useLoactionFeatureLoader = () => {
+  const fullyAttributedService = useFullyAttributedParcelMapLayer();
+  const adminBoundaryLayerService = useAdminBoundaryMapLayer();
+  const adminLegalBoundaryLayerService = useLegalAdminBoundariesMapLayer();
+
+  const {
+    loadProperties: { execute: loadProperties },
+  } = useMapProperties();
+
+  const fullyAttributedServiceFindOne = fullyAttributedService.findOne;
+  const adminBoundaryLayerServiceFindRegion = adminBoundaryLayerService.findRegion;
+  const adminBoundaryLayerServiceFindDistrict = adminBoundaryLayerService.findDistrict;
+  const adminLegalBoundaryLayerServiceFindOneMunicipality =
+    adminLegalBoundaryLayerService.findOneMunicipality;
+
+  const showLocationDetails = useCallback(
+    async (latLng: LatLngLiteral, isSelecting: boolean): Promise<LocationFeatureDataset> => {
+      const result: LocationFeatureDataset = {
+        location: latLng,
+        pimsFeature: null,
+        parcelFeature: null,
+        regionFeature: null,
+        districtFeature: null,
+        municipalityFeature: null,
+      };
+      try {
+        // call these APIs in parallel - notice there is no "await"
+        const fullyAttributedTask = fullyAttributedServiceFindOne(latLng);
+        const regionTask = adminBoundaryLayerServiceFindRegion(latLng, 'GEOMETRY');
+        const districtTask = adminBoundaryLayerServiceFindDistrict(latLng, 'GEOMETRY');
+
+        const parcelFeature = await fullyAttributedTask;
```

Comment:
Have a look at:
https://stackoverflow.com/questions/45285129/any-difference-between-await-promise-all-and-multiple-await/54291660#54291660

and see if you think that is relevant for this case.

Reply:
That's a good link, thanks for sharing it! I have updated this one and when I see future ones like these to follow that style as it looks cleaner (even we don't actually handle errors in this case)